src/xenia/config.cc

```diff
@@ -15,6 +15,14 @@ std::shared_ptr<cpptoml::table> ParseFile(const std::wstring& filename) {
     throw cpptoml::parse_exception(xe::to_string(filename) +
                                    " could not be opened for parsing");
   }
+  // since cpptoml can't parse files with a UTF-8 BOM we need to skip them
+  char bom[3];
+  file.read(bom, sizeof(bom));
```

Comment:
What if it reads less than 3 bytes?

Reply:
Then `file.fail()` in the next line returns `true` and the stream is reset to position 0 before passing it to cpptoml.